eox_nelp/course_experience/tests/tests_models.py

```diff
@@ -0,0 +1,86 @@
+"""This file contains all the test for models.py file.
+
+Classes:
+    FeedbackCourseTestCase: Test FeedbackCourse model.
+    FeedbackUnitTestCase: Test FeedbackUnit model.
+"""
+import unittest
+
+from django.contrib.auth import get_user_model
+from mock import patch
+
+from eox_nelp.course_experience.models import FeedbackCourse, FeedbackUnit
+from eox_nelp.edxapp_wrapper.course_overviews import CourseOverview
+
+User = get_user_model()
+
+
+class BaseFeedbackTestCase:
+    """Base class for feedback models."""
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Setup common conditions for every test case"""
+        self.author, _ = User.objects.get_or_create(
+            username="fake-user",
+            email="fake@example.com",
+        )
+        self.course, _ = CourseOverview.objects.get_or_create(id="course-v1:test+Cx108+2024_T4")
+
+    @patch("eox_nelp.course_experience.models.tracker")
+    def test_event_is_emitted(self, tracker_mock):
+        """
+        Tests that the feedback event is emmitted when an instance model is saved
+
+        Expected behavior:
+            - emit method is called with the right data.
+        """
+        expected_event_data = {
+            key: str(value)
+            for key, value in self.instance_data.items()  # pylint: disable=no-member
+        }
+        expected_event_data["author"] = str(self.author.id)
+        instance = self.test_class(**self.instance_data)  # pylint: disable=no-member
+
+        instance.save()
+
+        tracker_mock.emit.assert_called_once_with(
+            self.event_name,  # pylint: disable=no-member
+            expected_event_data,
+        )
+
+
+class FeedbackCourseTestCase(BaseFeedbackTestCase, unittest.TestCase):
+    """Test class for FeedbackCourse model. """
+
+    def setUp(self):
+        """Setup common conditions for every test case"""
+        super().setUp()
+        self.test_class = FeedbackCourse
+        self.instance_data = {
+            "author": self.author,
+            "rating_content": 5,
+            "feedback": "This is a great course",
+            "public": False,
+            "course_id": self.course,
+            "rating_instructors": 4,
+            "recommended": True,
```

Comment:
Here is also `time` in my local_env, would that key be also in tests?
![image](https://github.com/eduNEXT/eox-nelp/assets/51926076/8f2f58e5-2391-4977-8b72-205773c2b41d)

Reply:
time is outside event data, if you take a look you will realize that there is  `},`  before the time key